src/averroes/soot/CodeGenerator.java

```diff
@@ -515,6 +549,11 @@ private void callFinalize() {
 	 */
 	private void callApplicationMethodsReflectively() {
 		for (SootMethod toCall : getAllMethodsToCallReflectively()) {
+			// TODO: Is that correct? We want to skip lifecycle methods because these are modeled in the dummy main
+			if (isLifeCycle(toCall)) {
```

Comment:
Hmm, why is that support added to callApplicationMethodsReflectively? I'd rather this be added to the proper method and the list of lifecycle methods to be retrieved from the proper list of methods to call.